server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala

```diff
@@ -462,28 +462,35 @@ class ${className} extends (() => Any) with java.io.Serializable {
     val classPath        = getClassPath(this.getClass.getClassLoader)
     val currentClassPath = classPath.head
 
+    def checkCurrentClassPath: List[String] = currentClassPath match {
+      case List(jarFile) if jarFile.endsWith(".jar") =>
+        val relativeRoot = new File(jarFile).getParentFile
+        val jarResource: Resource[IO, JarFile] =
+          Resource.make(IO(new JarFile(jarFile)))(jar => IO(jar.close()))
+
+        val nestedClassPath: IO[String] =
+          jarResource
+            .use(jar => IO(jar.getManifest.getMainAttributes.getValue("Class-Path")))
+            .handleError { throwable: Throwable =>
+              throw new CompilerException(List(List(throwable.getMessage)))
+            }
```

Comment:
Following Fede's suggestion after my PR:

```scala
            .handleError { case scala.util.control.NonFatal(e) =>
              throw new CompilerException(List(List(e.getMessage)))
            }
```

Reply:
I have a question here, what is doing here the `scala.util.control.NonFatal(e)`?

Reply:
https://www.scala-lang.org/api/2.12.10/scala/util/control/NonFatal$.html

Reply:
I'm almost sure is not necessary for the cats `ApplicativeError` methods (they already avoid handling fatal exceptions) but it's ok to have it.